### DISCUSSION TITLE
getPublicUrl faster on server or client?

### DISCUSSION THREAD

**Discussion by IHIutch** [supabase/supabase#3778] [Questions] [answered]
I'm wondering about the fastest way to handle loading a feed of images (using supabase storage), something similar to an Instagram feed. I could `map` a list of images on the server or the client (are there benefits one way or the other?) Or, I'm also thinking that if I handled getting the image URLs on the client, I might be able leverage a lazy load like pattern.

Has anyone implemented something similar to this and have suggestions?

**Comment by dshukertjr** [comment, marked as answer]
Hi @IHIutch!

If your images are going to be public, I would just store the image URL directly on a database. 

For example, if you have a `posts` table like the following, I would just have `image_url` as one of the columns. This way, you will only have to get the URL once, and can easily leverage lazy loading or caching. 

| id | image_url                                                           | description |
|----|---------------------------------------------------------------------|-------------|
| 1  | https://sample.supabase.co/storage/v1/object/public/posts/file1.jpg | Lorem       |
| 2  | https://sample.supabase.co/storage/v1/object/public/posts/file2.jpg | ipsum       |
| 3  | https://sample.supabase.co/storage/v1/object/public/posts/file3.jpg | dolor       |

**Reply by IHIutch** [reply]
Ah I see. So since they're public, I can just fetch the publicURL when I initially store them and then save the public URL instead of fetching it every time I need them?

That seems like a no-brainer, lol. Thanks for helping me realize I was over-complicating things! 😅

**Reply by dshukertjr** [reply]
@IHIutch Yeah, exactly 🎉

**Reply by ghost** [reply]
Hello, how to upload from react native, i mean how to add post name to the image url?

**Reply by rcapdepaula** [reply]
> Hi @IHIutch!
> 
> If your images are going to be public, I would just store the image URL directly on a database.
> 
> For example, if you have a `posts` table like the following, I would just have `image_url` as one of the columns. This way, you will only have to get the URL once, and can easily leverage lazy loading or caching.
> 
> id	image_url	description
> 1	https://sample.supabase.co/storage/v1/object/public/posts/file1.jpg	Lorem
> 2	https://sample.supabase.co/storage/v1/object/public/posts/file2.jpg	ipsum
> 3	https://sample.supabase.co/storage/v1/object/public/posts/file3.jpg	dolor

Hi! How do you throw the uploaded image URL to your field image_url  in your table?
